components/segmenter/src/complex/dictionary.rs

```diff
@@ -137,13 +137,13 @@ impl<'l, 's> DictionaryType<'l, 's> for char {
     }
 }
 
-pub(crate) struct DictionarySegmenter<'l> {
+pub(super) struct DictionarySegmenter<'l> {
     dict: &'l UCharDictionaryBreakDataV1<'l>,
     grapheme: &'l RuleBreakDataV1<'l>,
 }
 
 impl<'l> DictionarySegmenter<'l> {
-    pub fn new(
+    pub(super) fn new(
         dict: &'l DataPayload<UCharDictionaryBreakDataV1Marker>,
```

Comment:
Nit (optional): Can you use this PR to address my unresolved comment from #3348

> Nit: You should almost never have a
> 
> `&'a DataPayload<FooV1Marker>`
> 
> you should instead have a
> 
> `&'a FooV1<'a>`

Reply:
I don't think this is the place